autoload/suda.vim

```diff
@@ -25,11 +26,20 @@ function! suda#system(cmd, ...) abort
   endif
   try
     call inputsave()
-    redraw | let password = inputsecret('Password: ')
+    redraw
+    if !sudo_needs_a_password
+      let password = inputsecret('Password: ')
+    else
+      let password = ''
+    end
   finally
     call inputrestore()
   endtry
-  let cmd = printf('sudo -p '''' -S %s', a:cmd)
+  if !sudo_needs_a_password
+    let cmd = printf('sudo -p '''' -S %s', a:cmd)
+  else
+    let cmd = printf('sudo -p '''' -n %s')
```

Comment:
missing `, a:cmd` and the command is exactly same as the command executed at line 19 so it would fail again.

Reply:
`sudo_needs_a_password` is `String` so it should checked with `==#` or whatever.

Reply:
Agree, forgot that.